File: February/CPP/Sequential Digits/Sequential Digits.cpp
class Solution {
    void helper(int curr, int low, int high, int cd, int hd){
        if(cd> hd || curr> high)
            return;
        if (curr <= high && curr>= low){
            ans.insert(curr);
        }
        int last = curr%10;
        if( last == 9)
            return;
        curr = curr *10 +(last+1);
        helper(curr,low,high,cd+1,hd);
    }

public:
    set<int> ans;
    vector<int> sequentialDigits(int low, int high) {
        vector<int> resl;
       int dl =0, dh =0, n=low, m=high;
       while(n>0){
           n/=10;
           dl++;
       }
       while(m>0){
           m/=10;
           dh++;
       }
       for(int i =0;i<9;i++)
        helper(i,low,high,0,dh);
       // sort(ans.begin(),ans.end());
       // return ans;
       for(int x: ans){
           resl.push_back(x);
       }
        return resl;
    }
};

File: February/Java/Sequential Digits/Sequential Digits.java
class Solution {
    public List<Integer> sequentialDigits(int low, int high) {
        int start = 1, cur = 1, inc = 11, t = 8;
        List<Integer> ans = new ArrayList<>();
        while(cur <= high){
            for (int i = 0; i < t; i++){
                cur += inc;
                if (cur >= low && cur <= high) ans.add(cur);
            }
            inc = 10 * inc + 1;
            start = 10 * start + 10 - t;
            cur = start;
            t--;
        }
        return ans;
    }
}
